src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -336,7 +335,6 @@ public function configureOptions(OptionsResolver $resolver)
         $resolver->setAllowedTypes('choice_list', array('null', 'Symfony\Component\Form\ChoiceList\ChoiceListInterface'));
         $resolver->setAllowedTypes('choices', array('null', 'array', '\Traversable'));
         $resolver->setAllowedTypes('choice_translation_domain', array('null', 'bool', 'string'));
-        $resolver->setAllowedTypes('choices_as_values', 'bool');
```

Comment:
not tech diff as you said, and since this line is going to be removed anyway in 3.1, there is no need to keep it